### PR TITLE
Print path to error if settings aren't valid

### LINKED ISSUE #1429: include path to invalid property in settings validation messages
SublimeLinter just upgraded and is now giving me the following completely unhelpful message:

image

I have no idea what config key this is referring to, and so have no way of fixing it.

### PR DESCRIPTION
**Pull request by kaste**
Fixes #1429

### CODE REVIEW

**Comment by kaste** [comment]
@braver 👍 from you?

**Comment by braver** [comment]
I haven’t tested, and can’t right now, but looking st the code I’m not sure what the message ends up looking like?

**Comment by kaste** [comment]
It's super simple. E.g.

```
'kill_old_processes': '' is not of type 'boolean'

OR

'linters'->'annotations'->'styles'->'mark_style': 'stippled_underlines' is not one of ['fill', 'outline', 'solid_underline', 'squiggly_underline', 'stippled_underline', 'none']
```

File: lint/settings.py [hunk 1]
```diff
@@ -86,18 +85,27 @@ def validate_settings():
...
             try:
-                validate(settings, schema)
+                validate(settings, schema, format_checker=FormatChecker())
             except ValidationError as error:
                 good = False
-                error_msg = error.message.split("\n")[0]  # reduce verbosity
-                full_msg = "Invalid settings in '{}':\n{}".format(name, error_msg)
-
-                logger.error(full_msg)
+                path_to_err = ('->'.join(
```
Comment (braver) [inline lint/settings.py]: I'd prefer ` > ` to join with visually, or `.` because that's JS syntax.

Comment (kaste) [reply lint/settings.py]: `.`s are out bc our keys can contain points or dashes. So the JSON is not a JS object.

I merge with `>` but if you can come up with a really good formatting, just write it down. For now, I still drop array indices so it print `styles>icon` but styles is a list and we don't tell the user which style definition failed. (Although the validator knows.) But `styles > 0 > icon` seemed ugly.

Comment (braver) [reply lint/settings.py]: Hm, if it’s not a JS (-like) path (which would be my ideal), it’s good like this. Actually, we should use a different character than `&gt;`, I’ll look it up later.

**Pull request merged**